services/admin/src/server.pins.test.ts

```diff
@@ -0,0 +1,41 @@
+import { Application } from 'express';
+import request from 'supertest';
+import { dirSync } from 'tmp';
+
+import { generatePin } from '@votingworks/utils';
+import { mockOf } from '@votingworks/test-utils';
+
+import { buildApp } from './server';
+import { createWorkspace, Workspace } from './util/workspace';
+
+jest.mock('@votingworks/utils', (): typeof import('@votingworks/utils') => {
+  return {
+    ...jest.requireActual('@votingworks/utils'),
+    generatePin: jest.fn(),
+  };
+});
+
+let app: Application;
+let workspace: Workspace;
+
+beforeEach(() => {
+  mockOf(generatePin).mockReset();
+
+  workspace = createWorkspace(dirSync().name);
```

Comment:
With our recent battles with `/tmp` disk space usage I've been trying to be more aware of when we're not cleaning things up. Add `fs.rm(workspace.path, { recursive: true })` to an `afterEach`?